--- configure.ac~	2024-12-27 15:58:57
+++ configure.ac	2024-12-27 16:48:23
@@ -4,11 +4,9 @@
 # Define mod_tile version number
 m4_define([mod_tile_version], [0.7.2])
 
-AC_PREREQ([2.61])
+AC_PREREQ([2.72])
 AX_CONFIG_NICE
-AC_INIT([mod_tile],
-        [mod_tile_version],
-        [http://trac.openstreetmap.org])
+AC_INIT([mod_tile],[mod_tile_version],[http://trac.openstreetmap.org])
 AM_INIT_AUTOMAKE([subdir-objects])
 LT_INIT
 AC_CONFIG_SRCDIR([src/mod_tile.c])
@@ -18,14 +16,13 @@
 # Checks for programs.
 AC_PROG_CXX
 AC_PROG_CC
-AC_PROG_CC_C99
 
 dnl Find C++ compiler
-AC_CHECK_PROG(HAVE_CXX, $CXX, yes, no)
-if test "$HAVE_CXX" = "no"
-then
-  AC_MSG_ERROR([Could not find a c++ compiler]);
-fi
+dnl AC_CHECK_PROG(HAVE_CXX, $CXX, yes, no)
+dnl if test "$HAVE_CXX" = "no"
+dnl then
+dnl   AC_MSG_ERROR([Could not find a c++ compiler]);
+dnl fi
 
 # Checks for libraries.
 PKG_CHECK_MODULES([GLIB], [glib-2.0])
@@ -46,14 +43,14 @@
 AC_FUNC_MKTIME
 
 AX_PTHREAD(,[AC_MSG_ERROR([no])])
-AC_SEARCH_LIBS(socket, socket)
-AC_SEARCH_LIBS(inet_ntoa, nsl)
-AC_SEARCH_LIBS(gethostbyname, resolv nsl)
-AC_SEARCH_LIBS(pow,m)
-AC_SEARCH_LIBS(clock_gettime,[rt posix4])
-PKG_CHECK_MODULES([MAPNIK], [libmapnik >= 4], [
+AC_SEARCH_LIBS([socket], [socket])
+AC_SEARCH_LIBS([inet_ntoa], [nsl])
+AC_SEARCH_LIBS([gethostbyname], [resolv nsl])
+AC_SEARCH_LIBS([pow], [m])
+AC_SEARCH_LIBS([clock_gettime], [rt posix4])
+PKG_CHECK_MODULES([MAPNIK], [libmapnik >= 4],[
      AC_DEFINE([HAVE_MAPNIK], [1])
-     AC_SUBST([MAPNIK_LDFLAGS], [${MAPNIK_LIBS}])
+     AC_SUBST([MAPNIK_LDFLAGS], [MAPNIK_LIBS])
     ], [AX_LIB_MAPNIK]
 )
 AX_ENABLE_LIBMEMCACHED
